[Global Nav] Moved Overview page/link to the accordion/Solution title

cchaos opened the pull request:
## Overview Pages

Initially we did not implement this because not all solutions had an "Overview" page. Now they do. So instead of listing every solution's overview page individually, they are now linked to through the text of the Solution's section in the global navigation. This still allows room for those wanting to collapse the item, but ensure the text itself links to the solution (and doesn't toggle the accordion).

![image](https://user-images.githubusercontent.com/549577/139107750-bf620158-2ab3-47ee-ac4a-a71321845018.png)

This change was also vetted through FullStory by finding that average time between successive clicks of the group's heading (the accordion toggle) was only `1s`. This roughly reveals that clicking the accordion header did not do what they expected, which is more likely to actually navigate to that solution.

<img width="1075" alt="Screen Shot 2021-10-01 at 14 01 07 PM" src="https://user-images.githubusercontent.com/549577/138751585-9e42f00f-c384-4ddb-84c9-aebe37eea747.png">

This PR completely filters out any links named "Overview" and passes that to the accordion title instead. The current approach isn't i18n-friendly as it looks for the exact word "Overview". There's definitely a better way to solve this, but I don't know where or how the plugins establish these links, I just know what they come in as.

Here's the full list. It cleans/reduces it quite a bit. 

![image](https://user-images.githubusercontent.com/549577/139107563-add4505b-eeba-4de3-b75c-975fd94ab697.png)

## Deployment link

I've also updated the "Manage this Deployment" link to not use the left arrow (which is confusing because it actually takes you to an entirely different platform) but to use the Cloud logo. And changes the background color to "dark" to highlight it. 

<img width="317" alt="Screen Shot 2021-10-25 at 14 55 21 PM" src="https://user-images.githubusercontent.com/549577/138753896-e280a988-afe8-4d1e-88f6-c9bfd7c2fec2.png">



### Checklist

Delete any items that are not applicable to this PR.

- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [x] Any UI touched in this PR is usable by keyboard only (learn more about [keyboard accessibility](https://webaim.org/techniques/keyboard/))
- [x] Any UI touched in this PR does not create any new axe failures
- [x] This was checked for [cross-browser compatibility](https://www.elastic.co/support/matrix#matrix_browsers)

Commit:
[Global Nav] Moved Overview page/link to the accordion/Solution title

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/158212/)
* Commit: f002052c0f13d129fb8b665fce4f60818ffd02b7
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Execute xpack-kibana-ciGroup2](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/158212/execution/node/600/log)
* [Execute xpack-kibana-ciGroup7](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/158212/execution/node/680/log)

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/158212/","number":"158212","commit":"f002052c0f13d129fb8b665fce4f60818ffd02b7"}]}
PIPELINE-->

Commit:
Using Cloud logo for deployment link and fixing isActive state

Commit:
Revert custom DTS

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/158244/)
* Commit: 5e94ef2ba2d15493d5c429d2dcb0a1e6681a1229
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check Doc API Changes](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/158244/execution/node/425/log)

### History
* :broken_heart: [Build #158212](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/158212/) failed f002052c0f13d129fb8b665fce4f60818ffd02b7

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/158212/","number":"158212","commit":"f002052c0f13d129fb8b665fce4f60818ffd02b7"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/158244/","number":"158244","commit":"5e94ef2ba2d15493d5c429d2dcb0a1e6681a1229"}]}
PIPELINE-->

Commit:
Fix prop

kibanamachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/158248/)
* Commit: 4bc4c61e275e640b5457c9a7452e143d7885802d
* Storybooks not built
* [Pipeline Steps](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/158248/flowGraphTable) (look for red circles / failed steps)
* [Interpreting CI Failures](https://www.elastic.co/guide/en/kibana/current/interpreting-ci-failures.html)

### History
* :broken_heart: [Build #158244](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/158244/) failed 5e94ef2ba2d15493d5c429d2dcb0a1e6681a1229
* :broken_heart: [Build #158212](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/158212/) failed f002052c0f13d129fb8b665fce4f60818ffd02b7

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/158212/","number":"158212","commit":"f002052c0f13d129fb8b665fce4f60818ffd02b7"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/158244/","number":"158244","commit":"5e94ef2ba2d15493d5c429d2dcb0a1e6681a1229"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/158248/","number":"158248","commit":"4bc4c61e275e640b5457c9a7452e143d7885802d"}]}
PIPELINE-->

kibanamachine committed:
Merge branch 'master' into global_nav/overview_pages

Commit:
Change Deployment section to dark background

Commit:
Completely remove locking ability

Commit:
Remove “Overview” links from tests

Commit:
Remove overview links from `appsMenu` test service

kibanamachine committed:
Merge branch 'master' into global_nav/overview_pages

Commit:
i18n

Commit:
Change buttonElement if overview page exists

Commit:
Snaps from button/div

ryankeairns reviewed (approved):
Nice use of FullStory as an input!

smith reviewed (approved):
APM changes look good

kibanamachine committed:
Merge branch 'master' into global_nav/overview_pages

kibanamachine committed:
Merge branch 'master' into global_nav/overview_pages

dominiqueclarke reviewed (approved):
Uptime changes LGTM

nreese reviewed (approved):
maps changes LGTM
code review

estermv reviewed (approved):
Infra Monitoring UI changes LGTM!

kibanamachine committed:
Merge branch 'master' into global_nav/overview_pages

pgayvallet reviewed (approved):
LGTM when the follow-up issue discussed in https://github.com/elastic/kibana/pull/114018#discussion_r737577199 is created

Commit:
PR Feedback:

- Puts `Overview` links back for Observability and Security
- Increases hit area and scopes underline of anchor tag within the accordion header
- Uses `id`’s to find Overview pages instead of `title` text
- Updates tests

Commit:
🤦‍♀️Remove ids from labels

cchaos reviewed (commented):
All: I've added the "Overview" links back in only for Observability and Security (while still linking to them in every solution's accordion header). I've updated the screenshot in the PR summary.

kibanamachine committed:
Merge branch 'master' into global_nav/overview_pages

Commit:
🤦‍♀️And the snaps

formgeist reviewed (approved):
👍  from Observability UX

kibanamachine committed:
Merge branch 'master' into global_nav/overview_pages

crob611 reviewed (approved):
Presentation Changes ✅

cchaos commented:
@elasticmachine merge upstream

cchaos commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into global_nav/overview_pages

cchaos commented:
@elastic/kibana-data-discovery I'm going to merge this at the EOD today. Your CodeOwner file is just fixing a test to update the list of nav items that removes "Overview".

kibanamachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/3285)
* Commit: cb42f29b650c6fb6d853ecbe48ffeb4ac8c4a16c
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-114018/cb42f29b650c6fb6d853ecbe48ffeb4ac8c4a16c)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/3285#1d3e911c-2acb-4520-8b8c-4c224e52716f) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/3285/jobs/1d3e911c-2acb-4520-8b8c-4c224e52716f/artifacts/5d33b6ce-caa6-41d1-b917-e81b2c80c6bc) Docker CI Group / Fleet Endpoints EPM Endpoints package installation error handling and rollback on a fresh install, it should uninstall a broken package during rollback

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/12af55889864b7578d792be6fb5e9d7488842ef2) | [after](https://github.com/elastic/kibana/commit/cb42f29b650c6fb6d853ecbe48ffeb4ac8c4a16c) | diff |
| --- | --- | --- | --- |
| `core` | 265.9KB | 266.3KB | +398.0B |



### History
* :green_heart: [Build #2578](https://buildkite.com/elastic/kibana-pull-request/builds/2578) succeeded 3a2d91658f92c3447f2f963ea720892f14d36b51
* :broken_heart: [Build #2183](https://buildkite.com/elastic/kibana-pull-request/builds/2183) failed 48b35f6168a71e665efc03ef18f415891da78d88
* :broken_heart: [Build #2146](https://buildkite.com/elastic/kibana-pull-request/builds/2146) failed e73e73cea703ef7eb76fc646c03febe8cff4f2ea
* :green_heart: [Build #1661](https://buildkite.com/elastic/kibana-pull-request/builds/1661) succeeded 3f1ffa68b53f75a38eea8804bf0bbd3691b7ad4c
* :broken_heart: [Build #1618](https://buildkite.com/elastic/kibana-pull-request/builds/1618) failed 43a975e0572fc3529fbaafd025d4d1b7fdab58f0
* :broken_heart: [Build #1568](https://buildkite.com/elastic/kibana-pull-request/builds/1568) failed 687dd175560eb5a35be26de5da773f79cc669ed5

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/3285","number":3285,"commit":"cb42f29b650c6fb6d853ecbe48ffeb4ac8c4a16c"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/2578","number":2578,"commit":"3a2d91658f92c3447f2f963ea720892f14d36b51"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/2183","number":2183,"commit":"48b35f6168a71e665efc03ef18f415891da78d88"},{"buildStatus":{"state":"running","success":false,"hasRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/2146","number":2146,"commit":"e73e73cea703ef7eb76fc646c03febe8cff4f2ea"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/1661","number":1661,"commit":"3f1ffa68b53f75a38eea8804bf0bbd3691b7ad4c"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/1618","number":1618,"commit":"43a975e0572fc3529fbaafd025d4d1b7fdab58f0"}],"number":3285}
buildkite-pr-comment-->

kibanamachine commented:
The following labels were identified as gaps in your version labels and will be added automatically:
- v8.1.0

If any of these should not be on your pull request, please manually remove them.

kibanamachine commented:
## 💚 Backport successful

| Status | Branch | Result |
|:------:|:------:|:------:|
| ✅ |  [8.0](https://github.com/elastic/kibana/pull/117137)  | [<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/117137">](https://github.com/elastic/kibana/pull/117137) |

This backport PR will be merged automatically after passing CI.
